### ISSUE TITLE
Lab 6 Level 3

### ISSUE THREAD

**Issue by joshlimwe** [nus-cs2030/2223-s2#599]
# Summary

Hello, how come i need to redefine the method "put" in both student and module even after I defined it in my KeyableMap class.  If i were to put an assignment into a module using my code below, wouldnt V be an assignment and KeyableMap<V> be a module? Thank you!

## Description:

Redefining put method

## Code Snippets

<img width="289" alt="image" src="https://user-images.githubusercontent.com/125873745/227714977-0099f8f4-eb3b-4898-8a87-a9cc496bd062.png">

**Comment by ianwonghanli** [comment]
For the KeyableMap<V> class' put function, you are returning a KeyableMap<V> class. However for Student class and Module class, the put functions in both classes need to return a Student class and Module class respectively. Hence, there is a need to redefine the put function.

**Comment by tristantanjh** [comment]
Yes, like the others have commented, Student takes in a Module when calling the put function, and not a KeyableMap<Assessment>

**Comment by NwaterM** [comment]
Basically, you have a general V in the put method of keyableMap, but you will need to specify what is the type of object you are putting into Student and Module respectively. For Student, we put a Module in, but for Module, we put an Assessment in, so you need to redefine your put <Types> in their classes. Hope it helps :))

**Comment by WXUXIN** [comment]
In the KeyableMap class' put function, you are returning a KeyableMap class. But in Student and Module, you are returning another Student and Module respectively

**Comment by viethaaaa** [comment]
hi you will need to override the put method in the generic class with a new put method that specifies the return type to be Student/Module instead of KeyableMap
